### PR TITLE
[WIP] Add an option for new users to have private visibility

### PR DESCRIPTION
**Pull request by brianhelba**

### CODE REVIEW

**Comment by brianhelba** [comment]
This supports a use case for @satra . PTAL.

**Comment by brianhelba** [comment]
@girder/developers To fully support this, since `User` is an `AccessControlledModel` type, I think we should also expose a `/user/:id/access` endpoint, and add a `girder.views.AccessWidget` to the `girder.views.UserAccountView`. Does this sound reasonable?

**Comment by zachmullen** [comment]
There are certain related scenarios I'd want to test, like what happens if I'm a group admin and a user belongs to that group but I don't have permission to view the user?

File: clients/web/src/templates/body/systemConfiguration.jade [hunk 1]
```diff
@@ -77,6 +77,21 @@ form.g-settings-form(role="form")
...
+          selected=(
+            settings['core.user_default_public'] === null ?
+            defaults['core.user_default_public'] === true :
+            settings['core.user_default_public'] === true)
+          ) Public
+        option(value="false",
+          selected=(
+            settings['core.user_default_public'] === null ?
+            defaults['core.user_default_public'] === false :
+            settings['core.user_default_public'] === false)
```
Comment (zachmullen) [inline clients/web/src/templates/body/systemConfiguration.jade]: Maybe we should capture this structure in a function so we don't have to repeat it

**Comment by brianhelba** [comment]
@zachmullen I can tell you the answer from my own experience using private users: the private user actually shows up in the group list to everyone (even anonymous users). But you're right, this should be tested further.

**Comment by satra** [comment]
thanks @brianhelba 

@zachmullen - i think this speaks to the granularity of access and visibility more generally. brian and i had a bit of a discussion over file level access permissions as well. i think there is a place for a broader discussion of acl's amongst groups, users, collections, items, and files. just noting it here for now, but happy to move to a new issue for discussion.

**Comment by zachmullen** [comment]
@brianhelba what's the status of this PR, are you comfortable that everything regarding private users works properly?

**Comment by jcfr** [comment]
@brianhelba This was closed during today's hackfest because it didn't see activity for a while, feel free to re-open or comment. Thanks

**Pull request closed without merging**